File: editorial/chap02/cpp/answer_B07.cpp
#include <iostream>
using namespace std;

int N, T;
int L[500009], R[500009];
int Answer[500009], B[500009];

int main() {
	// 입력
	cin >> T >> N;
	for (int i = 1; i <= N; i++) cin >> L[i] >> R[i];

	// 전일 비에 더한다
	for (int i = 0; i <= T; i++) B[i] = 0;
	for (int i = 1; i <= N; i++) {
		B[L[i]] += 1;
		B[R[i]] -= 1;
	}

	// 누적 합을 구한다
	Answer[0] = B[0];
	for (int d = 1; d <= T; d++) Answer[d] = Answer[d - 1] + B[d];

	// 출력
	for (int d = 0; d < T; d++) cout << Answer[d] << endl;
	return 0;
}

File: editorial/chap02/python/answer_B07.py
# 입력
T = int(input())
N = int(input())
L = [ None ] * N
R = [ None ] * N
for i in range(N):
	L[i], R[i] = map(int, input().split())

# "이전 시간과의 차이"에 추가
B = [ 0 ] * (T + 1)
for i in range(N):
	B[L[i]] += 1
	B[R[i]] -= 1

# 누적 합을 구한다
Answer = [ None ] * (T + 1)
Answer[0] = B[0]
for d in range(1, T+1):
	Answer[d] = Answer[d-1] + B[d]

# 출력
for d in range(T):
	print(Answer[d])
